projects/packages/sync/src/class-sender.php

```diff
@@ -529,6 +539,10 @@ public function get_items_to_send( $buffer_or_items, $encode = true ) {
 				$skipped_items_ids[] = $key;
 				continue;
 			}
+			if ( $this->action_is_full_sync( $item[0] ) && strlen( wp_json_encode( $item ) ) > self::MAX_SIZE_FULL_SYNC ) {
```

Comment:
I initially thought of doing this check after encoding, and comparing it against `$upload_size` instead but I did get some `Fatal error: Maximum execution time of 30 seconds exceeded` when testing.

Reply:
That's strange. I agree it's best to check after encoding and avoid encoding the item twice. Let's figure out and fully understand why this results in Fatal